cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -1614,6 +1619,9 @@ macro(build_thrift)
   if(DEFINED BOOST_ROOT)
     list(APPEND THRIFT_CMAKE_ARGS "-DBOOST_ROOT=${BOOST_ROOT}")
   endif()
+  if(DEFINED Boost_INCLUDE_DIR)
+    list(APPEND THRIFT_CMAKE_ARGS "-DBoost_INCLUDE_DIR=${Boost_INCLUDE_DIR}")
```

How about using `BOOST_INCLUDE_DIR` instead of `Boost_INCLUDE_DIR` because `BOOST_...` is a hint variable https://cmake.org/cmake/help/latest/module/FindBoost.html#hints and `Boost_...` is a cache result variable https://cmake.org/cmake/help/latest/module/FindBoost.html#cache-variables ?

```suggestion
    list(APPEND THRIFT_CMAKE_ARGS "-DBOOST_INCLUDE_DIR=${Boost_INCLUDE_DIR}")
```

---

I passed the result var as it short circuits a load of logic in findboost which was breaking. I suspect that's why the namespace is passed similarly.